test-utils/src/main/java/com/criteo/publisher/csm/ConcurrentSendingQueueHelper.java

```diff
@@ -16,10 +16,14 @@
 
 package com.criteo.publisher.csm;
 
+import static org.mockito.Mockito.mockingDetails;
+
 public class ConcurrentSendingQueueHelper {
 
   @SuppressWarnings("KotlinInternalInJava")
   public static void emptyQueue(ConcurrentSendingQueue<?> queue) {
-    queue.poll(Integer.MAX_VALUE);
+    if (!mockingDetails(queue).isMock() || mockingDetails(queue).isSpy()) {
```

Comment:
What's the idea behind verifying that queue isn't a mock? Do you want to override `thenCallRealMethod`?

Reply:
The idea is to not add an interaction on a mock. Without this condition, a test was broken because it was doing a `verifyNoInteraction(injectedQueue)`.

A true solution would be to either:
- clean all mock interactions after setup just before the test
- (*) have more finer control on when an action should be done

In this case the method (*) is locally apply to sending queues: it is not necessary to clean it when it is a mock.
I would say that for the moment, something like this is sufficient. If one day we start having issues because of interactions in the setup, then we could think about a solution.